client/src/menu/game/gameScreenContent/ChatMenu.tsx

```diff
@@ -3,189 +3,157 @@ import translate from "../../../game/lang";
     
     const self = useRef<HTMLDivElement>(null);
 
     const AT_BOTTOM_THRESHOLD_PIXELS = 40;
-
-    const filteredMessages = useMemo(() => {
-        if (props.filter === null) return messages;
-        else return messages.filter(msg => props.filter?.test(translateChatMessage(msg, GAME_MANAGER.getPlayerNames())) || msg.type === "phaseChange")
-    }, [messages, props.filter]);
-
     const handleScroll = (e: any) => {
         const { scrollTop, scrollHeight, clientHeight } = e.target;
         setScrolledToBottom(scrollTop + clientHeight >= scrollHeight - AT_BOTTOM_THRESHOLD_PIXELS);
     }
-
-
-
-    // Keep chat scrolled to bottom
+    
+    
+    const [filter, setFilter] = useState<PlayerIndex | null>(null);
     useEffect(() => {
-        if (scrolledToBottom && self.current !== null) {
-            const el = self.current;
-            el.scrollTop = el.scrollHeight;
+        const stateListener: StateListener = (type) => {
+            if (GAME_MANAGER.state.stateType === "game") {
+                setFilter(GAME_MANAGER.state.chatFilter);
+            }
         }
-    }, [self, messages, props.filter, scrolledToBottom])
+        GAME_MANAGER.addStateListener(stateListener);
+        return () => GAME_MANAGER.removeStateListener(stateListener);
+    }, [setFilter]);
 
     // Update with new messages
     useEffect(() => {
         const stateListener: StateListener = (type) => {
             if (GAME_MANAGER.state.stateType === "game" && type === "addChatMessages") {
-                setMessages(GAME_MANAGER.state.chatMessages.filter(msg => 
-                    props.filter === null || msg.type === "phaseChange" || props.filter.test(translateChatMessage(msg, GAME_MANAGER.getPlayerNames()))
-                ))
+                setMessages(GAME_MANAGER.state.chatMessages)
             }
         }
 
         GAME_MANAGER.addStateListener(stateListener);
         return () => GAME_MANAGER.removeStateListener(stateListener);
-    }, [props.filter, setMessages]);
+    }, [filter, setMessages]);
+
+    // Keep chat scrolled to bottom
+    useEffect(() => {
+        if (scrolledToBottom && self.current !== null) {
+            const el = self.current;
+            el.scrollTop = el.scrollHeight;
+        }
+    }, [self, messages, filter, scrolledToBottom])
 
     //scroll chat to bottom when filter is shut off or loaded
     useEffect(() => {
         if (self.current === null) return;
         self.current.scrollTop = self.current.scrollHeight;
-    }, [props.filter])
+    }, [filter])
 
     
 
     return <div className="message-section" ref={self} onScroll={handleScroll}>
         <div className="message-list">
-            {filteredMessages.map((msg, index) => {
+            {messages.filter((msg)=>{
+                const msgtxt = translateChatMessage(msg, GAME_MANAGER.getPlayerNames());
+                return filter === null || msg.type === "phaseChange" || msgtxt.includes(GAME_MANAGER.getPlayerNames()[filter]);
```

Comment:
We talked in discord but I'll also put it here

It's probably better to have it so if someone doesn't put a space before or after the name it still finds it because people are not perfect and will do that (me) and you still want to see the messages that that person is sending

Reply:
Valid